### PR TITLE
Feat/155294

### LINKED ISSUE #155294: Allow me to disable debug status bar treatment
I am becoming a power-user of our debugger (yeah, finally). This means that most of my "VS Code time" I have some kind of debuggee running or connected. So, I am seeing the ugly orange debug status bar treatment a lot. I don't like it and for me it's enough to see the activitybar icon decorations. So, pretty please give me an option to disable this

### PR DESCRIPTION
**Pull request by antonioprudenzano**
This PR fixes #155294 and  adds the ability to disable the coloured status bar when in debug mode.

I resumed the work left behind by @KoushikSahu and added the onDidChangeConfiguration listener.

### CODE REVIEW

**Comment by antonioprudenzano** [comment]
> @antonioprudenzano please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company=""]

**Comment by antonioprudenzano** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

**Comment by antonioprudenzano** [comment]
Gentle ping, @roblourens, any feedback?

File: src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts [hunk 1]
```diff
@@ -61,15 +62,25 @@ export class StatusBarColorProvider implements IWorkbenchContribution {
...
+		this.configurationService.onDidChangeConfiguration((e) => {
+			if (e.affectsConfiguration('debug.enableStatusBarColor')) {
+				this.update();
+			}
+		});
 		this.update();
 	}
 
 	protected update(): void {
 		this.enabled = isStatusbarInDebugMode(this.debugService.state, this.debugService.getModel().getSessions());
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts:79]: Can you write this method to only set `this.enabled` once per call?

Comment (antonioprudenzano) [reply src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts:79]: the update() method?

Comment (antonioprudenzano) [reply src/vs/workbench/contrib/debug/browser/statusbarColorProvider.ts]: yeah, fixed

**Review by roblourens — APPROVED** [review]
Perfect, thank you!

**Review by joyceerhl — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**